plugins/database/mongodb/connection_producer.go

```diff
@@ -78,6 +81,32 @@ func (c *mongoDBConnectionProducer) Connection() (interface{}, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	if c.WriteConcern != "" {
+		input := c.WriteConcern
+
+		// Try to base64 decode the input. If successful, consider the decoded
+		// value as input.
+		inputBytes, err := base64.StdEncoding.DecodeString(input)
+		if err == nil {
+			input = string(inputBytes)
+		}
+
+		var concern writeConcern
+		err = json.Unmarshal([]byte(input), &concern)
```

Comment:
This might be better to do in the Initialize function and cache the `writeConcern` object in the struct. Will save on marshaling it on every call.

Reply:
This only gets through if `Connection()` creates a new session, otherwise it will return pretty early up top. Since other session settings were set here, I thought calling `SetSafe` here as well.

Reply:
It's also probably better to return any unmarshaling errors during Init too.

Reply:
I'd say decode and unmarshal in Init, store the `mgo.Safe` pointer in the struct and call `SetSafe` in `Connection()`